quill-engine/src/main/scala/io/getquill/norm/AttachToEntity.scala

```diff
@@ -69,8 +69,9 @@ object AttachToEntity {
       case ConcatMap(IsEntity(a), b, c) => ConcatMap(f(a, b), b, c)
       case Filter(IsEntity(a), b, c) => Filter(f(a, b), b, c)
       case SortBy(IsEntity(a), b, c, d) => SortBy(f(a, b), b, c, d)
+      case DistinctOn(IsEntity(a), b, c) => DistinctOn(f(a, b), b, c)
 
-      case Map(_: GroupBy, _, _) | _: Union | _: UnionAll | _: Join | _: FlatJoin => f(q, alias.getOrElse(Ident("x", q.quat)))
+      case Map(_: GroupBy, _, _) | Map(_: DistinctOn, _, _) | _: Union | _: UnionAll | _: Join | _: FlatJoin => f(q, alias.getOrElse(Ident("x", q.quat)))
```

Comment:
I don't think `Map(_: DistinctOn, _, _) | ` is needed here. Please try to remove it.